document.go

```diff
@@ -0,0 +1,43 @@
+package opensearchtools
+
+import "encoding/json"
+
+// RoutableDoc interface defines an OpenSearch document that can be routed to a specific index.
+// Most documents will route to a single index. A basic implementation might look like:
+//
+//	type BasicDoc struct {
+//		ID 		string
+//		Index 	string
+//	}
+//
+//	func (b *BasicDoc) ID() string {
+//		return b.ID
+//	}
+//
+//	func (b *BasicDoc) Index() string{
+//		return b.Index
+//	}
+type RoutableDoc interface {
```

Comment:
Is `RoutableDoc` still necessary?  Maybe we leave it but also include a default/standard implementation?
```
func NewDoc(id, index string) Doc {
   return Doc{
      id: id,
      index: index,
   }
}
type Doc struct {
   id, index string
}
func (doc Doc) ID() string {
   return doc.id
}
func (doc Doc) Index() string {
   return doc.index
}
```

Something else I noticed ... the example here is invalid now because a type can't have both an `ID string` field and an `ID() string` method.  Same for `Index string` and `Index() string`.

Reply:
Ah, good catch on the example. I certainly can do `BasicDoc` as a default implementation.

It's not needed for this iteration of the mget, but still useful for the write side of operations.

Reply:
created `DocumentID` as that basic implementation.

Reply:
👍 Need to add `func NewDocumentID(id, index string) DocumentID` though.

Reply:
doh, added.